client/my-sites/stats/stats-interval-dropdown/index.jsx

```diff
@@ -1,49 +1,81 @@
 import { Button, Dropdown } from '@wordpress/components';
 import { check, Icon, chevronDown } from '@wordpress/icons';
-import React, { useEffect, useState } from 'react';
-import Intervals from 'calypso/blocks/stats-navigation/intervals';
+import page from 'page';
+import qs from 'qs';
 import './style.scss';
 
-const IntervalDropdown = ( { period, pathTemplate } ) => {
-	const [ currentInterval, setCurrentInterval ] = useState( period );
-
-	useEffect( () => {
-		setCurrentInterval( period );
-	}, [ period ] );
-
+const IntervalDropdown = ( { slug, period, queryParams } ) => {
 	const intervalLabels = {
 		day: 'Days',
 		week: 'Weeks',
 		month: 'Months',
 		year: 'Years',
 	};
 
-	const getCurrentIntervalLabel = ( intervalValue ) => {
-		return intervalLabels[ intervalValue ] || intervalValue;
-	};
+	// New interval listing that preserves date range.
+	// TODO: Figure out how to dismiss on select.
+
+	function generateNewLink( newPeriod ) {
+		const newRangeQuery = qs.stringify( Object.assign( {}, queryParams, {} ), {
+			addQueryPrefix: true,
+		} );
+		const url = `/stats/${ newPeriod }/${ slug }`;
+		return `${ url }${ newRangeQuery }`;
+	}
+
+	function getDisplayLabel() {
```

Comment:
I do appreciate the attempt to make the code blocks more readable. However it's a tricky balance, isn't it? IMHO, we wouldn't want to wrap up a single line of object value access in a function, because it's already pretty clear. 🤔